### PR TITLE
feat(api): Support 96 channel in the hardware controller

### PR DESCRIPTION
**Pull request by Laura-Danielle**
# Overview

This branch includes work to get the 96 channel supported in the hardware controller. I would like to merge it in after we deliver any updates to the science robot since I did not have a chance to test all pipette types last week in the office.

# Changelog

- [feat(ot3): add shared data support for ot3 pipettes (](https://github.com/Opentrons/opentrons/commit/352fdfd3232916b5461594cde22facc10ae0f5e4)
- [feat(shared-data): properly load new pipette configurations from shared data](https://github.com/Opentrons/opentrons/commit/e59f558908144eaea5a6646b0da519abeff45a10)
- [feat(api): Support the new pipette configurations and 96 channel in the hardware controller](https://github.com/Opentrons/opentrons/commit/3a3035b888219fce60dabcf8517be342075d1984) 
- [feat(api): change pipette functions based on the tip working volume](https://github.com/Opentrons/opentrons/commit/3a3035b888219fce60dabcf8517be342075d1984)
- [feat(api): do not use "pipette name" and "pipette model" to look up pipettes via configurations](https://github.com/Opentrons/opentrons/commit/3a3035b888219fce60dabcf8517be342075d1984)

# Review requests

Definitely comment if you see anything glaring.

# Risk assessment
Medium/High. Modifies configurations for the OT-3 across the pipette models

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
lets do it!

File: api/src/opentrons/hardware_control/backends/ot3controller.py [hunk 1]
```diff
@@ -403,21 +402,43 @@ async def gripper_home_jaw(self) -> None:
             self._homed_nodes.add(axis)
 
     @staticmethod
-    def _synthesize_model_name(name: FirmwarePipetteName, model: str) -> "PipetteModel":
-        return cast("PipetteModel", f"{name.name}_v{model}")
+    def _lookup_serial_key(pipette_name: FirmwarePipetteName) -> str:
```
Comment (ahiuchingau) [inline api/src/opentrons/hardware_control/backends/ot3controller.py:444]: IMO lookup tables such as this should be in pipette config or the ot3utils. Is there a reason why they have to be here?

File: api/src/opentrons/hardware_control/backends/ot3controller.py [hunk 2]
```diff
@@ -403,21 +402,43 @@ async def gripper_home_jaw(self) -> None:
...
+            FirmwarePipetteName.p1000_multi: "P1KM",
+            FirmwarePipetteName.p50_single: "P50S",
+            FirmwarePipetteName.p50_multi: "P50M",
+            FirmwarePipetteName.p1000_96: "P1KH",
+            FirmwarePipetteName.p50_96: "P50H",
+        }
+        return lookup_name[pipette_name]
+
+    @staticmethod
+    def _combine_serial_number(pipette_info: ohc_tool_types.PipetteInformation) -> str:
```
Comment (ahiuchingau) [inline api/src/opentrons/hardware_control/backends/ot3controller.py:456]: Nice, we'll use this for gripper as well

Also can be in ot3utils, maybe?

**Review by ahiuchingau — APPROVED** [review]
Looks good to me, just got two nits/questions

File: api/src/opentrons/hardware_control/backends/ot3controller.py (hunk 1 above)
Comment (Laura-Danielle) [reply api/src/opentrons/hardware_control/backends/ot3controller.py:444]: The real fix is just to return the full serial number from `PipetteInformation`, so I would consider both of these functions to be temporary.

File: api/src/opentrons/hardware_control/backends/ot3controller.py (hunk 2 above)
Comment (Laura-Danielle) [reply api/src/opentrons/hardware_control/backends/ot3controller.py:456]: Yeah I made the change for the gripper as well, but see my answer to your comment above. The serial # is only used for purposes of identifying the instrument and saving calibration, so I think we should just make sure we return the full "serial number" from `PipetteInformation` when an instrument is first detected.

File: api/src/opentrons/hardware_control/backends/ot3controller.py (hunk 1 above)
Comment (ahiuchingau) [reply api/src/opentrons/hardware_control/backends/ot3controller.py:444]: Could you leave a comment/todo so we know they are temp?

**Pull request merged**